hermes-consumers/src/test/groovy/pl/allegro/tech/hermes/consumers/consumer/sender/http/headers/HermesHeadersProviderTest.groovy

```diff
@@ -65,19 +71,51 @@ class HermesHeadersProviderTest extends Specification {
     def "should forward headers from nested provider"() {
         given:
         HttpHeadersProvider nestedHeadersProvider = new HttpHeadersProvider() {
+
             @Override
-            HttpRequestHeaders getHeaders(Message message) {
-                return new HttpRequestHeaders(Collections.singletonMap("k", "v"))
+            HttpRequestHeaders getHeaders(Message message, String rawAddress) {
+                return new HttpRequestHeaders(singletonMap("k", "v"))
             }
         }
 
-        HttpHeadersProvider hermesHeadersProvider = new HermesHeadersProvider(nestedHeadersProvider)
+        HttpHeadersProvider hermesHeadersProvider = new HermesHeadersProvider(
+                singleton(nestedHeadersProvider))
 
         when:
-        Map<String, String> headers = hermesHeadersProvider.getHeaders(message()).asMap()
+        Map<String, String> headers = hermesHeadersProvider.getHeaders(message(), rawAddress()).asMap()
 
         then:
         headers.get("k") == "v"
     }
 
+    def "should produce headers appending headers with additional providers"() {
+        def additionalProviderOne = new HttpHeadersProvider() {
```

Comment:
the "given:" section should start here IMO :]

Reply:
True ;)